python/lsst/daf/butler/tests/_testRepo.py

```diff
@@ -137,9 +140,11 @@ def makeTestCollection(repo: Butler) -> Butler:
     isolated test area, and not for repositories intended for real data
     processing or analysis.
     """
-    # Create a "random" collection name
-    # Speed matters more than cryptographic guarantees
-    collection = "test" + str(random.randrange(1_000_000_000))
+    if not uniqueId:
+        # Create a "random" collection name
+        # Speed matters more than cryptographic guarantees
+        uniqueId = str(random.randrange(1_000_000_000))
+    collection = "test" + uniqueId
```

Comment:
I'd put an underscore or other character between the `test` and the id, to make it easier to read (on the rare occasion when a person needs to read it).